### PR TITLE
chore: change conda pkg manager to mamba

### LINKED ISSUE #18: feat: replace snakemake pkg manager: conda -> mamba
Since we use _snakemake_ version > `6.0.0` the preferable frontend package manager for building internal, per-rule envs is now [mamba](https://github.com/mamba-org/mamba), not _conda_.

I have tried to tacke it quickly but, strangely enough, when I specify to use _mamba_ then our output files of _cutadapt_ do not match the expected checksums... (see: https://github.com/zavolanlab/zarp/pull/16#issuecomment-931562637). Therefore for now our profiles specify `--conda-frontend conda`.

It might be nice to replace _conda_ with _mamba_, eventually...

### LINKED ISSUE #52: bug: local test with conda fails?
**Describe the bug**
Running `bash tests/test_integration_workflow_with_conda/test.local.sh` exits with 
```
WorkflowError:
Python package jinja2 must be installed to create reports.
  File "/home/christina/miniconda3/envs/zarp/lib/python3.7/site-packages/snakemake/report/__init__.py", line 594, in auto_report
```
This happens at the step 
```
+ snakemake --snakefile=../../workflow/Snakefile --configfile=../input_files/config.yaml --report=snakemake_report.html
```
although the environment.yml for zarp conda env specifies ` jinja2=2.11.2`

**To Reproduce**
1. Follow installation instructions in zarp `README` to create the conda env (I did a fresh creation of the conda env root & update with dev)
2. `conda activate zarp`
3. run `bash tests/test_integration_workflow_with_conda/test.local.sh`

**Expected behavior**
exit status 0. 

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by AngryMaciek**
## Description

Fixes #18 

Strangely enough the previous problems with `cudadapt` output inconsistencies are no longer there. Maybe they have updated their build at the Anaconda Cloud(?)

Also Fixes #52, which we discovered by the way.

CC @ninsch3000 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation updated

## Checklist:

- [x] My code changes follow the style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] I have updated the project's documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +48,22 @@ git clone https://github.com/zavolanlab/zarp.git
...
 for your system (Linux). Be sure to select Python 3 option. 
 The workflow was built and tested with `miniconda 4.7.12`.
 Other versions are not guaranteed to work as expected.
 
+Given that Miniconda has been installed and is available in the current shell the first
+dependency for ZARP is the [Mamba][mamba] package manager, which needs to be installed in
+the `base` conda environment with:
+
+```bash
+conda install mamba -n base -c conda-forge
```
Comment (fgypas) [inline README.md:64]: That's good that you it mention here. I think that we should apply the same methodology in the tests. If we switch to mamba we will speed up a lot the testing process.

**Comment by fgypas** [comment]
Since this is an important issue, did you test also on the cluster?

File: README.md (hunk 1 above)
Comment (AngryMaciek) [reply README.md:64]: Now, upon the new push - yes, docs updated ;)

Comment (AngryMaciek) [reply README.md:64]: Update: `mamba env update` would not work, reverted that change, see https://github.com/mamba-org/mamba/issues/631

**Comment by AngryMaciek** [comment]
> Since this is an important issue, did you test also on the cluster?

Yes, cluster run is OK too

**Review by mkatsanto — APPROVED** [review]

**Pull request merged**